gensim/models/sldamodel.py

```diff
@@ -0,0 +1,223 @@
+import logging
+import numbers
+import os
+
+import numpy as np
+import six
+from scipy.special import gammaln, psi  
+from scipy.special import polygamma
+from six.moves import xrange
+from collections import defaultdict
+
+from gensim import interfaces, utils, matutils
+from gensim.matutils import dirichlet_expectation
+from gensim.models import basemodel, CoherenceModel
+
+try:
+    from scipy.special import logsumexp
+except ImportError:
+    from scipy.misc import logsumexp
+    
+logger = logging.getLogger('gensim.models.sldamodel')
+
+def update_dir_prior(prior, N, logphat, rho):
+    """
+    Updates a given prior using Newton's method, described in
+    **Huang: Maximum Likelihood Estimation of Dirichlet Distribution Parameters.**
+    http://jonathan-huang.org/research/dirichlet/dirichlet.pdf
+    """
+    dprior = np.copy(prior)  # TODO: unused var???
```

Comment:
??

Reply:
I had a feeling that there is an unused parameter for the ordering, so I put in the comment. I can remove it if needed.